cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -131,7 +132,7 @@ type sourcegraphClient struct {
 }
 
 func (s *sourcegraphClient) List(ctx context.Context, indexed []uint32) (*SourcegraphListResult, error) {
-	repos, err := s.listRepoIDs(ctx, indexed)
+	reposIter, repos, err := s.listRepoIDs(ctx, indexed)
```

Comment:
The changes in `List` and `s.listRepoIDs` are probably the core of the whole change.